### PR TITLE
Updated dwds (and other packages)

### PR DESCRIPTION
**Pull request by annagrin**
## Description

DWDS update includes a bug fix where native JavaScript objects are
not included in list of variables available in scopes.
The bug was causing the IntelliJ dart extension to throw exception
while trying to show 'this' variable in static functions, and VSCode
dart extension to display the native javascript objects with error
fields inside.

## Related Issues

See: https://github.com/dart-lang/webdev/issues/965

## Tests

Added test to verify that 'this' does not appear in scope of a static function in:
`flutter_tools/test/integration.shard/expression_evaluation_web_test.dart`

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I signed the [CLA].
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] All existing and new tests are passing.
- [x] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Did any tests fail when you ran them? Please read [Handling breaking changes].

- [x] No, no existing tests failed, so this is *not* a breaking change.
- [ ] Yes, this is a breaking change. *If not, delete the remainder of this section.*
   - [ ] I wrote a design doc: https://flutter.dev/go/template *Replace this with a link to your design doc's short link*
   - [ ] I got input from the developer relations team, specifically from: *Replace with the names of who gave advice*
   - [ ] I wrote a migration guide: *Replace with a link to your migration guide*

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Test Coverage]: https://github.com/flutter/flutter/wiki/Test-coverage-for-package%3Aflutter
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Handling breaking changes]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes

### CODE REVIEW

File: packages/flutter_tools/lib/src/build_runner/resident_web_runner.dart [hunk 1]
```diff
@@ -525,8 +525,6 @@ class _ResidentWebRunner extends ResidentWebRunner {
       }
     } on Exception catch (err) {
       return OperationResult(1, err.toString(), fatal: true);
-    } on WipError catch (err) {
```
Comment (devoncarew) [inline packages/flutter_tools/lib/src/build_runner/resident_web_runner.dart:528]: 👍 I'd seen this when trying local upgrades of `package: webkit_inspection_protocol`.

**Review by devoncarew — APPROVED** [review]
lgtm (though, I haven't looked at the CI failures)

**Comment by devoncarew** [comment]
The CI failures look related to package version solving issues.

**Comment by devoncarew** [comment]
I would try rev'ing this dep: `https://github.com/flutter/flutter/blob/master/dev/integration_tests/simple_codegen/pubspec.yaml#L5` to `1.3.0` as an explicit edit to the pubspec file.

I'm not sure why it's pegged to a specific version; looks like the last edit was by @jonahwilliams.

**Comment by jonahwilliams** [comment]
IIRC there were some temporary breakages with the web builds, but that should be resolved by now. It should be safe to remove the constraint and re-run update-packages

**Comment by jonahwilliams** [comment]
The integration test failures look legitimate though:

https://cirrus-ci.com/task/6626285176750080

Seems like something changed in package:vm_service. I also mentioned this to @bkonyi and @zanderso

**Comment by bkonyi** [comment]
Can you reproduce this locally? If so, can you try updating [this line](https://github.com/flutter/flutter/blob/9b7b9d795edfdd12c6ad8dd6495bed789c1bac05/packages/flutter_tools/test/integration.shard/vmservice_integration_test.dart#L43) in the test to pass in `<String, dynamic>{}` for the args? It looks like there might be an issue where invocations with no parameters default to using a `<dynamic, dynamic>{}` as the params if there were no params passed.

**Comment by annagrin** [comment]
> Can you reproduce this locally? If so, can you try updating [this line]

I passed the isolate id and it worked (because it effectively creates the proper args). Looks like if the following line gets executed, the empty args map is created in some calls downstream that has `Map` type instead of `Map<String, dynamic>`:

https://github.com/dart-lang/sdk/blob/3aa7484f6a61b5d04365075ab4f0371cc50deb9e/pkg/vm_service/lib/src/vm_service.dart#L1958

File: packages/flutter_tools/lib/src/build_runner/resident_web_runner.dart (hunk 1 above)
Comment (annagrin) [reply packages/flutter_tools/lib/src/build_runner/resident_web_runner.dart:528]: That's because WipError implements exception now. Analyzer rules:)

**Comment by bkonyi** [comment]
`package:vm_service` v4.0.4 is released. Try updating to that version to hopefully fix the failures you're seeing.

**Review by jonahwilliams — APPROVED** [review]
LGTM

**Comment by Zazo032** [comment]
@annagrin this PR is causing me some problems after doing a clean install of Flutter:
```
C:\src>flutter --version
Checking Dart SDK version...
Downloading Dart SDK from Flutter engine b59e3e9c39a252402fb535cacff12855fad13972...
Unzipping Dart SDK...
Building flutter tool...
Running pub upgrade...
Because flutter_tools depends on devtools 0.2.5 which doesn't match any versions, version solving failed.
Error (1): Unable to 'pub upgrade' flutter tool. Retrying in five seconds... (9 tries left)
```
And it keeps trying until this appears:
```
Warning: The following flags were passed as VM options and are being ignored: --packages=C:\src\flutter\packages\flutter_tools\.packages
These flags should be passed after the dart command (e.g., 'dart run --enable-asserts foo.dart' instead of 'dart --enable-asserts run foo.dart').
Could not find a command named "C:\src\flutter\bin\cache\flutter_tools.snapshot".

Usage: dart [<vm-flags>] <command|dart-file> [<arguments>]

Global options:
-h, --help       Print this usage information.
-v, --verbose    Show verbose output.
    --version    Print the Dart SDK version.

Available commands:
  analyze   Analyze the project's Dart code.
  create    Create a new project.
  format    Format Dart source code.
  migrate   Perform a null safety migration on a project or package.
  pub       Work with packages.
  run       Run a Dart file.
  test      Runs tests in this project.

Run "dart help <command>" for more information about a command.
```
It's weird because `devtools 0.2.5` does indeed match a published version: https://pub.dev/packages/devtools/versions/0.2.5

**Comment by bkonyi** [comment]
> And it keeps trying until this appears:
> 
> ```
> Warning: The following flags were passed as VM options and are being ignored: --packages=C:\src\flutter\packages\flutter_tools\.packages
> These flags should be passed after the dart command (e.g., 'dart run --enable-asserts foo.dart' instead of 'dart --enable-asserts run foo.dart').
> Could not find a command named "C:\src\flutter\bin\cache\flutter_tools.snapshot".
> 
> Usage: dart [<vm-flags>] <command|dart-file> [<arguments>]
> 
> Global options:
> -h, --help       Print this usage information.
> -v, --verbose    Show verbose output.
>     --version    Print the Dart SDK version.
> 
> Available commands:
>   analyze   Analyze the project's Dart code.
>   create    Create a new project.
>   format    Format Dart source code.
>   migrate   Perform a null safety migration on a project or package.
>   pub       Work with packages.
>   run       Run a Dart file.
>   test      Runs tests in this project.
> 
> Run "dart help <command>" for more information about a command.
> ```

Hm, this particular failure is probably related to some DartDev changes. I'll submit a PR to disable it for now.

**Comment by annagrin** [comment]
@Zazo032 @bkonyi does this issue still exist? I tried a fresh install and could not repro the problem. Let me know if it still persists and what are your install steps.

**Comment by bkonyi** [comment]
I'm not sure about the devtools version error, but the second error should be resolved by the PR above.

**Comment by Zazo032** [comment]
@annagrin this is still happening to me:
```
C:\src>git clone https://github.com/flutter/flutter.git -b master
Cloning into 'flutter'...
remote: Enumerating objects: 236405, done.
remote: Total 236405 (delta 0), reused 0 (delta 0), pack-reused 236405Receiving objects: 100% (236405/236405), 92.60 MiB | 18.87 MiB/s
Receiving objects: 100% (236405/236405), 98.85 MiB | 16.69 MiB/s, done.
Resolving deltas: 100% (181011/181011), done.
Updating files: 100% (4903/4903), done.

C:\src>flutter --version
Checking Dart SDK version...
Downloading Dart SDK from Flutter engine b59e3e9c39a252402fb535cacff12855fad13972...
Unzipping Dart SDK...
Building flutter tool...
Running pub upgrade...
Because flutter_tools depends on devtools 0.2.5 which doesn't match any versions, version solving failed.
Error (1): Unable to 'pub upgrade' flutter tool. Retrying in five seconds... (9 tries left)
```
It may be solved by the PR mentioned above, I'll try again when it's merged

**Comment by Zazo032** [comment]
Just tried @bkonyi's `disable-dart-dev` but that issue still appears, not sure why:
```
C:\src>git clone https://github.com/flutter/flutter.git -b disable-dart-dev
Cloning into 'flutter'...
remote: Enumerating objects: 236405, done.
remote: Total 236405 (delta 0), reused 0 (delta 0), pack-reused 236405
Receiving objects: 100% (236405/236405), 98.85 MiB | 13.23 MiB/s, done.
Resolving deltas: 100% (181013/181013), done.
Updating files: 100% (4903/4903), done.

C:\src>flutter --version
Checking Dart SDK version...
Downloading Dart SDK from Flutter engine b59e3e9c39a252402fb535cacff12855fad13972...
Unzipping Dart SDK...
Building flutter tool...
Running pub upgrade...
Because flutter_tools depends on devtools 0.2.5 which doesn't match any versions, version solving failed.
Error (1): Unable to 'pub upgrade' flutter tool. Retrying in five seconds... (9 tries left)
```

**Comment by bkonyi** [comment]
@Zazo032 yeah, that change wouldn't fix your pub issues. There is a [devtools 0.2.5](https://pub.dev/packages/devtools) on pub so it looks like you might just be in a bad state. Have you tried deleting your `.pubcache` and trying it again?

**Comment by Zazo032** [comment]
Managed to fix it by deleting Pub's folder

**Pull request merged**